futures-util/src/compat/compat01as03.rs

```diff
@@ -163,13 +163,13 @@ mod io {
         fn poll_flush(&mut self, waker: &task03::Waker)
             -> task03::Poll<Result<(), Error>>
         {
-            poll_01_to_03(self.in_notify(waker, |x| x.poll_flush()))
+            poll_01_to_03(self.in_notify(waker, AsyncWrite01::poll_flush))
```

Comment:
IMO it's cleaner to have these match above-- I think this clippy lint seems a bit overzealous, but also not particularly worth arguing over...